server/functions/src/who.ts

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable */
+
+// This was generated from who.proto with https://github.com/stephenh/ts-proto
+// Then all of the encoding methods were removed as they interfered with
+// Firestore and are not necessary.
+// This leaves only the interface definitions.
```

Comment:
I like this a lot! I in particular like how the use of these interfaces produces very nice self-documenting code in the other `.ts` files.

Reply:
However it needs better tooling to keep in sync w/proto changes. Probably just some flags to ts-proto but I have not investigated.